sources/src/parser/gmlobjectparser.cpp

```diff
@@ -1,22 +1,26 @@
 #include "parser/gmlobjectparser.h"
 
 #include <citygml/object.h>
+#include "parser/attributes.h"
+#include <citygml/citygmlfactory.h>
+#include "parser/documentlocation.h"
+
+#include <iostream>
 
 namespace citygml {
 
     GMLObjectElementParser::GMLObjectElementParser(CityGMLDocumentParser& documentParser, CityGMLFactory& factory, std::shared_ptr<CityGMLLogger> logger)
-        : CityGMLElementParser(documentParser, factory, logger)
-    {
-
+        : CityGMLElementParser(documentParser, factory, logger) {
+        m_lastCodeSpace = "";
+        m_attributeHierarchy = std::shared_ptr<std::vector<AttributesMap>>(new std::vector<AttributesMap>);
```

Comment:
sharedポインタの初期化は`make_shared`で
また、メンバの初期化は初期化子リストの使用をお願いいたします。
```suggestion
        : CityGMLElementParser(documentParser, factory, logger),
          m_attributeHierarchy(std::make_shared<std::vector<AttributesMap>>()) {
```